File: omnes.Web/Modules/ServerTypes/Parametros/TiposTelefonoRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TiposTelefonoRow {
    IdTipoTelefono?: number;
    NombreTipoTelefono?: string;
}

export abstract class TiposTelefonoRow {
    static readonly idProperty = 'IdTipoTelefono';
    static readonly nameProperty = 'NombreTipoTelefono';
    static readonly localTextPrefix = 'Parametros.TiposTelefono';
    static readonly deletePermission = 'SuperAdmin';
    static readonly insertPermission = 'SuperAdmin';
    static readonly readPermission = 'SuperAdmin';
    static readonly updatePermission = 'SuperAdmin';

    static readonly Fields = fieldsProxy<TiposTelefonoRow>();
}

File: omnes.Web/Modules/Parametros/TiposTelefono/TiposTelefonoRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace omnes.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("TiposTelefono")]
[DisplayName("Tipos Telefono"), InstanceName("Tipos Telefono")]
[ReadPermission("SuperAdmin")]
[ModifyPermission("SuperAdmin")]
public sealed partial class TiposTelefonoRow : Row<TiposTelefonoRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Tipo Telefono"), Identity, IdProperty]
        public int? IdTipoTelefono { get; set; }

        [DisplayName("Nombre Tipo Telefono"), Size(100), NotNull, QuickSearch, NameProperty]
        public string NombreTipoTelefono { get; set; }
    }
}